lib/surrounded/context.rb

```diff
@@ -135,6 +143,7 @@ def role_map
       end
 
       def map_roles(role_object_array)
+        self.class.detect_collisions role_object_array
```

Comment:
Yes, but I had a very frustrating time trying to get it to work. If I ``include NameCollisionDetector`` then ``on_name_collision`` is undefined. If I ``extend NameCollisionDetector`` then calling ``detect_collisions`` or ``self.detect_collisions`` is undefined. If I create a submodule of ``NameCollisionDetector`` which contains ``detect_collisions`` and automatically ``include`` it into the context class then ``@handler`` is undefined. I would appreciate some pointers here since I spent much time googling on this subject without much success...